R/readWQPdata.R

```diff
@@ -96,12 +97,50 @@
 #' startDate <- as.Date("2013-01-01")
 #' nutrientDaneCounty <- readWQPdata(countycode="US:55:025",startDate=startDate,
 #'                         characteristicType="Nutrient")
+#' secchi.names = c("Depth, Secchi disk depth", 
+#'                  "Depth, Secchi disk depth (choice list)", 
+#'                  "Secchi Reading Condition (choice list)", 
+#'                  "Secchi depth", 
+#'                  "Water transparency, Secchi disc")
+#' args <- list('startDateLo' = startDate, 
+#'              'startDateHi' = "2013-12-31", 
+#'               statecode="WI", 
+#'               characteristicName=secchi.names)
 #' 
-#'                         
+#' wqp.data <- readWQPdata(args)   
+#' 
+#' args_2 <- list('startDateLo' = startDate, 
+#'              'startDateHi' = "2013-12-31", 
+#'               statecode="WI", 
+#'               characteristicName=secchi.names,
+#'               querySummary=TRUE)
+#'
+#' wqp.summary <- readWQPdata(args_2) 
+#' 
+#' arg_3 <- list('startDateLo' = startDate, 
+#'              'startDateHi' = "2013-12-31")
+#' arg_4 <- list(statecode="WI", 
+#'               characteristicName=secchi.names,
+#'               querySummary=TRUE)
+#' wqp.summary <- readWQPdata(arg_3, arg_4) 
+#'                
 #' }
 readWQPdata <- function(..., zip=FALSE, querySummary=FALSE){
   
-  matchReturn <- list(...)
+  if(all(sapply(list(...), class) != "list")){
+    matchReturn <- list(...)
+  } else {
+    matchReturn <- c(...)
+    for(i in c("zip","querySummary")){
+      if(do.call(missing, list(i)) & i %in% names(matchReturn)){
+        assign(i, matchReturn[[i]])
+      }
+      
+      if(i %in% names(matchReturn)){
+        matchReturn <- matchReturn[-which(names(matchReturn) %in% i)]
+      }
+    }
+  }
```

Comment:
use some type of append_replace (like in gsplot) where zip and querySummary are added to the list. Need more logic for building URL in this method, but eliminates need for `assign()`